bats/tests/containers/wasm.bats

```diff
@@ -0,0 +1,149 @@
+# shellcheck disable=SC2030,SC2031
+# See https://github.com/koalaman/shellcheck/issues/2431
+# https://www.shellcheck.net/wiki/SC2030 -- Modification of output is local (to subshell caused by @bats test)
+# https://www.shellcheck.net/wiki/SC2031 -- output was modified in a subshell. That change might be lost
+
+load '../helpers/load'
+
+# Bundled shims are this version or newer.
+BUNDLED_VERSION=0.11.1
+# Manually managed versions intentionally use an older version
+# so we can verify that they still override the bundled version.
+MANUAL_VERSION=0.10.0
+
+local_setup() {
+    if using_containerd; then
+        skip "this test only works on moby right now"
+    fi
+}
+
+@test 'factory reset' {
+    factory_reset
+    rm -rf "$PATH_CONTAINERD_SHIMS"
+}
+
+@test 'start engine without wasm support' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+shim_version() {
+    local shim=$1
+    local version=$2
+
+    run rdctl shell "containerd-shim-${shim}-${version}" -v
+    assert_success || return
+    semver "$output"
+}
+
+@test 'verify spin shim is not installed on PATH' {
+    run shim_version spin v2
+    assert_failure
+    assert_output --partial "containerd-shim-spin-v2: not found"
+}
+
+hello() {
+    local shim=$1
+    local version=$2
+    local lang=$3
+    local port=$4
+    local internal_port=$5
+
+    # The '/' at the very end of the command is required by the container entrypoint.
+    ctrctl run -d \
```

Comment:
We don't seem to be cleaning up the containers at the end either? Are we relying on the next test running factory reset?

Reply:
Yes. Leaving containers running is related to not shutting down Rancher Desktop at the end of the test: it allows you to manually debug things when they fail.